### DISCUSSION TITLE
ERROR  Cannot restart nuxt:  'unknown' type in '/app/i18n/locales/en.ts'.

### DISCUSSION THREAD

**Discussion by gvanto** [nuxt-modules/i18n#3270] [Q&A]
Nuxt project info:                                                                                                                                                                       9:10:29 AM

------------------------------
- Operating System: Linux
- Node Version:     v20.18.0
- Nuxt Version:     3.14.0
- CLI Version:      3.15.0
- Nitro Version:    2.10.2
- Package Manager:  yarn@1.22.22
- Builder:          -
- User Config:      default
- Runtime Modules:  @pinia/nuxt@0.5.5, @nuxtjs/i18n@8.5.6
- Build Modules:    -
------------------------------

As per these docs: https://i18n.nuxtjs.org/docs/v8/guide/lazy-load-translations#basic-usage

I am trying the following,
in my i18n/locales/en.ts file I have:

```
import { useAPIFetch } from '@/composables/useAPIFetch.ts';

type TranslationType = {
	[key: string]: string;
}

export default (async (locale: string): Promise<TranslationType> => {
	const {data, error, status}
		= await useAPIFetch('/v1/getTranslations?language=' + locale);

	//return await data.value as TranslationType; // gives same error
	return data.value as TranslationType;
});

```

The useAPIFetch is as follows:
```
/**
 * https://stackoverflow.com/questions/72041740/how-to-set-global-api-baseurl-used-in-usefetch-in-nuxt-3/75870302#75870302
 */
import { useFetch } from 'nuxt/app';

type useFetchType = typeof useFetch;

export const useAPIFetch: useFetchType = (path, options = {}) => {
    const config = useRuntimeConfig();

    options.baseURL = config.public.apiBaseUrl;

    return useFetch(path, options);
}
```

But I keep getting following error:
ERROR  Cannot restart nuxt:  'unknown' type in '/app/i18n/locales/en.ts'.

Have tried multiple things (from claude and chatgpt), with no luck (?)

Using "export default defineI18nLocale" (as per the docs) also doesn't work, get different error, same as here: https://github.com/nuxt-modules/i18n/issues/2145

Any help much appreciated!

Full error:
ERROR  Cannot start nuxt:  'unknown' type in '/app/i18n/locales/en.ts'.
rcms-nuxt  |
rcms-nuxt  |   at genImportSpecifier (node_modules/@nuxtjs/i18n/dist/module.mjs:1682:11)
rcms-nuxt  |   at generateLocaleImports (node_modules/@nuxtjs/i18n/dist/module.mjs:1636:29)
rcms-nuxt  |   at node_modules/@nuxtjs/i18n/dist/module.mjs:1651:37
rcms-nuxt  |   at Array.forEach (<anonymous>)
rcms-nuxt  |   at generateLoaderOptions (node_modules/@nuxtjs/i18n/dist/module.mjs:1651:19)
rcms-nuxt  |   at genTemplate (node_modules/@nuxtjs/i18n/dist/module.mjs:1890:12)
rcms-nuxt  |   at setup (node_modules/@nuxtjs/i18n/dist/module.mjs:1933:20)
rcms-nuxt  |   at async normalizedModule (node_modules/@nuxt/kit/dist/index.mjs:2140:17)
rcms-nuxt  |   at async installModule (node_modules/@nuxt/kit/dist/index.mjs:2397:95)
rcms-nuxt  |   at async initNuxt (node_modules/nuxt/dist/index.mjs:4669:5)

**Comment by greeffjean** [comment]
Hi Gvanto ! 

I can be much more helpful if I was able to re-produce this error, but assuming the codebase is for company eyes only, Ill try my best by looking at the code: 

So I can see some mistakes already: 

1. In `'@/composables/useAPIFetch.ts'` , the return type for the function `useAPIFetch` is incorrect, I imagine this could be where your  error is coming from, `type useFetchType = typeof useFetch;` is of type `Function`, and `useAPIFetch` returns and `Object` , can you see there is no overlap here ?
2. Another place your error could be coming from is `i18n/locales/en.ts`, you are type casting here `return data.value as TranslationType;` , you are likely breaking the rules of TS, for instance `data.value` may have no overlap with `TranslationType` 
3. Also, TS infers values, I think its overkill using type casting, rather just type the return value of the `useAPIFetch` function properly.
4. Lastly I find and would recommend using a IDE TS linting plugin, it helps with catching errors during code creation and shows helpful error messages

**Reply by greeffjean** [reply]
Happy to hop on a call with you, you know how to reach me !
